subsys/bluetooth/host/audio/mpl.c

```diff
@@ -2301,31 +2415,44 @@ uint64_t current_track_id_get(void)
 
 void current_track_id_set(uint64_t id)
 {
-	/* This requires that we have the track with the given ID */
-	/* and can find it and switch to it. */
-	/* There is also a matter of what to do with the group, */
-	/* does the track have a corresponding group, or do we create one? */
+	struct mpl_group_t *group;
+	struct mpl_track_t *track;
 
 	BT_DBG_OBJ_ID("Track ID to set: ", id);
 
-	/* What we really want to do: Check that we have a track with this ID */
-	/* Set the track to the track with this ID */
-	/* Temporarily, we construct a track with this id, and use that */
-	/* We chain it into the current group */
-	track_dummy.id = id;
-	track_dummy.prev = pl.group->track;
-	track_dummy.next = pl.group->track->next;
-	pl.group->track->next->prev = &track_dummy;
-	pl.group->track->next = &track_dummy;
-	pl.group->track = &track_dummy;
-	pl.track_pos = 0;
-	media_proxy_pl_current_track_id_cb(id);
-	media_proxy_pl_track_changed_cb(); /* Todo: Spec says not to notify the client */
-				/* who set the track */
+	find_track_by_id(&pl, id, &group, &track);
+
+	if (group != NULL && track != NULL) {
+
+		if (pl.group != group) {
+			pl.group = group;
+			do_group_change_notifications(&pl);
+
+			/* Group change implies track change (even if same track in other group) */
+			pl.group->track = track;
+			do_track_change_notifications(&pl);
+
+		} else if (pl.group->track != track) {
+			pl.group->track = track;
+			do_track_change_notifications(&pl);
+		}
+	}
+
+	BT_DBG("Track not found");
```

Comment:
Isn't this logged always, even if the track is found?

Reply:
Added missing return